### PR TITLE
`--timeout` option for `start` command

### LINKED ISSUE #276: [2pt] Start timeout
I suggest adding `--timeout` option  to specify the time to wait for instance is started.

Now on `cartridge start -d` CLI waits for instance to send `READY=1` to notify socket.
And it can wait forever...

I sure that we need to stop it after timeout (by default, it's a `1m`).
But user can say `cartridge start -d --timeout 10m` that mean `I want to wait 10 minutes until my instance is started`.

### PR DESCRIPTION
**Pull request by dokshina**
Closes #276

### CODE REVIEW

File: cli/commands/start.go [hunk 1]
```diff
@@ -2,13 +2,28 @@ package commands
...
 	"github.com/tarantool/cartridge-cli/cli/running"
 )
 
+const (
+	defaultStartTimeout = 1 * time.Minute
+)
+
+var (
+	daemonize      bool
+	timeout        time.Duration
```
Comment (olegrok) [inline cli/commands/start.go:20]: `type Duration int64` that means that `timeout` is signed.

Could we pass some negative value?

Comment (dokshina) [reply cli/commands/start.go:20]: This option accepts value in the `00h4m0.5s` form, the negative value will be rejected by cobra (or, as I understood, by the `time.Duration` constructor).

Comment (dokshina) [reply cli/commands/start.go]: I've added error on negative timeout.

File: README.rst [hunk 2]
```diff
@@ -378,6 +378,12 @@ The following options (``[flags]``) are supported:
 * ``--name string`` defines the application name.
   By default, it is taken from the application rockspec.
 
+* ``--timeout string`` Time to wait for instance(s) start in background.
+  Can be specified in seconds or in the duration form (``72h3m0.5s``).
```
Comment (olegrok) [inline README.rst:382]: `72h3m0.5s`

`0.5s`

Is it really possible to specify a half of second?

File: cli/commands/common.go [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+		durationStr = fmt.Sprintf("%ds", seconds)
+	}
+
+	duration, err := time.ParseDuration(durationStr)
+	if err != nil {
+		return 0, err
+	}
+
+	if duration < 0 {
+		return 0, fmt.Errorf("Negative duration is specified")
```
Comment (olegrok) [inline cli/commands/common.go:35]: I suggest "negative timeout"

**Comment by olegrok** [comment]
Is it possible/reasonable to implement simple wrapper (as for spinner) that will run some command with timeout. It allows to generalize `timeout` logic not only for start. But for stop and another commands that could be implemented in future.

File: README.rst (hunk 2 above)
Comment (dokshina) [reply README.rst:382]: Yes, it's parsed to `500 * time.Milisecond`. (but I will check id additionally)

File: cli/commands/common.go (hunk 3 above)
Comment (dokshina) [reply cli/commands/common.go:35]: But this function is called `getDuration`.

**Comment by dokshina** [comment]
Yes, but there are some problems with running spinner in different goroutines. I'm planning to implement it in a big separated patch.

File: cli/commands/common.go (hunk 3 above)
Comment (olegrok) [reply cli/commands/common.go:35]: But user specifies it as "timeout" and don't know about "durations"

Comment (olegrok) [reply cli/commands/common.go:35]: But ok. May be you are right

**Comment by olegrok** [comment]
> Yes, but there are some problems with running spinner in different goroutines. I'm planning to implement it in a big separated patch.

Please file an issue.

File: README.rst (hunk 2 above)
Comment (dokshina) [reply README.rst:382]: Yes, it's parsed to `500ms`. I've checked.

**Pull request merged**